### PR TITLE
TRUNK-5817: Migrate web and webapp package tests to Junit 5 (StarupErrorFilterTest)

### PR DESCRIPTION
**Pull request by achilep**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5817

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 1]
```diff
@@ -11,41 +11,45 @@
...
 import org.springframework.mock.web.MockHttpServletRequest;
 
 /**
  * Tests {@link StartupErrorFilter}.
  */
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(Listener.class)
-@PowerMockIgnore({"com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "javax.management.*", "org.w3c.dom.*"})
-public class StartupErrorFilterTest {
+public class StartupErrorFilterTest extends BaseContextSensitiveTest {
```
Comment (sherrif10) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:29]: Thanks alot @achilep  for your great , there are some failing tests that are breaking the ci ,could you first run your new changes

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 2]
```diff
@@ -11,41 +11,45 @@
...
+
+	private StartupErrorFilter filter;
 	
-	@Before
+	private Throwable errorAtStartup ;
```
Comment (teleivo) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:33]: we dont need to have a field for this one

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 3]
```diff
@@ -11,41 +11,45 @@
...
+
+	@BeforeEach
 	public void setUp() {
-		mockStatic(Listener.class);
+		filter = new StartupErrorFilter();
+	}
+	
+	@AfterEach
+	public void reverterrorAtStartup() {
+		errorAtStartup = null;
```
Comment (teleivo) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:42]: please inline the variable

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 4]
```diff
@@ -11,41 +11,45 @@
...
+		Whitebox.setInternalState(Listener.class, "errorAtStartup", errorAtStartup);
+		
 	}
 	
 	@Test
 	public void getModel_shouldReturnAStartupErrorFilterModelContainingTheStartupError() {
 		
 		Throwable t = mock(Throwable.class);
-		when(Listener.getErrorAtStartup()).thenReturn(t);
+		Whitebox.setInternalState(Listener.class, "errorAtStartup", t);
```
Comment (teleivo) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:51]: I think we don't need to mock the throwable. You can just create any `Exception` as they are `Throwable`

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 5]
```diff
@@ -55,20 +59,18 @@ public void getModel_shouldReturnAStartupErrorFilterModelContainingTheStartupErr
 	@Test
 	public void skipFilter_shouldReturnTrueIfNoErrorHasOccuredOnStartup() {
 		
-		when(Listener.errorOccurredAtStartup()).thenReturn(false);
+		Whitebox.setInternalState(Listener.class, "errorAtStartup", false);
```
Comment (teleivo) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:62]: remove this line, you are setting the errors to null in the after each , so there are no errors by default for this test. the test will then pass.

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java [hunk 6]
```diff
@@ -55,20 +59,18 @@ public void getModel_shouldReturnAStartupErrorFilterModelContainingTheStartupErr
...
 		
-		assertTrue("should be true on start without error", filter.skipFilter(new MockHttpServletRequest()));
+		assertTrue(filter.skipFilter(new MockHttpServletRequest()), "should be true on start without error");
 	}
 	
 	@Test
 	public void skipFilter_shouldReturnFalseIfAnErrorHasOccuredOnStartup() {
 		
-		when(Listener.errorOccurredAtStartup()).thenReturn(true);
+		Whitebox.setInternalState(Listener.class, "errorAtStartup", true);
```
Comment (teleivo) [inline web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:71]: you will need to set the errors field to a Throwable (for example an Exception) for the getter to return true

File: web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java (hunk 2 above)
Comment (sherrif10) [reply web/src/test/java/org/openmrs/web/filter/startuperror/StartupErrorFilterTest.java:33]: Thanks for this @teleivo

**Comment by achilep** [comment]
thanks @teleivo

**Pull request merged**